crates/rpc/rpc-eth-api/src/core.rs

```diff
@@ -540,6 +550,60 @@ where
             .await?)
     }
 
+    /// Handler for: `eth_getTransactionBySenderAndNonce`
+    async fn transaction_by_sender_and_nonce(
+        &self,
+        sender: Address,
+        nonce: U64,
+    ) -> RpcResult<Option<RpcTransaction<T::NetworkTypes>>> {
+        trace!(target: "rpc::eth", ?sender, ?nonce, "Serving eth_getTransactionBySenderAndNonce");
+        let nonce = nonce.saturating_to::<u64>();
+
+        // Check the pool first
+        if let Some(tx) = LoadState::pool(self).get_transaction_by_sender_and_nonce(sender, nonce) {
+            let transaction = tx.transaction.clone().into_consensus();
+            return Ok(Some(reth_rpc_types_compat::transaction::from_recovered(transaction)))
+        }
+
+        // Check if the sender is a contract
+        if self.get_code(sender, None).await?.len() > 0 {
+            return Ok(None)
+        }
+
+        let highest = EthState::transaction_count(self, sender, None).await?.saturating_to::<u64>();
+
+        // If the nonce is higher or equal to the highest nonce, the transaction is pending or not
+        // exists.
+        if nonce >= highest {
+            return Ok(None)
+        }
```

Comment:
this is very similar to what we already have in otterscan, I'm fine with this for now, but I assume we could minimize this by changing how we enforce 

https://github.com/paradigmxyz/reth/blob/dba315ff8e5e5a9937936e3c3ba5808ff6940a2b/crates/rpc/rpc/src/otterscan.rs#L72-L72

will open an issue for this

Reply:
thanks for the kindness, I'll minimize the codebase in another PR :)